### PR TITLE
bsd: Fix eventfd broken logic

### PR DESCRIPTION
**Pull request by marysaka**
This commit fix eventfd logic being broken.

The following changes were made:
- EventFd IPC definition had argument inverted
- EventFd events weren't fired correctly
- Poll logic was wrong and unfinished for eventfd
- ~Reintroduce workaround from #3385 but in a safer way, and spawn 4
  threads~ This was proven to be too expensive and need more rethinking.

~With this and #3646, fix Pokemon Legends Arceus 1.1.1 with guest internet enabled.~
With this and some rework on IPC, this will fix Pokemon Legends Arceus 1.1.1 with guest internet enabled

### CODE REVIEW

**Comment by EmulationFanatic** [comment]
Fixes online play on Use Your Words (errors out trying to connect/hangs for minutes at a time on master 1.1.243)
image

**Comment by Thog** [comment]
Should be ready for review.

**Review by gdkchan — COMMENTED** [review]
I suppose
```cs
            // FIXME: We should support blocking operations.
            // Right now they can't be supported because it would cause the
            // service to lock up as we only have one thread processing requests.
            flags |= EventFdFlags.NonBlocking;
```
can be removed now?

File: Ryujinx.HLE/HOS/Services/ServerBase.cs [hunk 1]
```diff
@@ -61,8 +73,12 @@ public ServerBase(KernelContext context, string name, Func<IpcService> smObjectF
 
         private void AddPort(int serverPortHandle, Func<IpcService> objectFactory)
         {
-            _portHandles.Add(serverPortHandle);
-            _ports.Add(serverPortHandle, objectFactory);
+            {
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.HLE/HOS/Services/ServerBase.cs]: Forgot to push that clean up sorry

File: Ryujinx.HLE/HOS/Services/ServerBase.cs [hunk 2]
```diff
@@ -75,17 +91,87 @@ public void AddSessionObj(KServerSession serverSession, IpcService obj)
         }
 
         public void AddSessionObj(int serverSessionHandle, IpcService obj)
+        {
+            {
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.HLE/HOS/Services/ServerBase.cs]: Forgot to push that clean up sorry

File: Ryujinx.HLE/HOS/Services/ServerBase.cs [hunk 3]
```diff
@@ -75,17 +91,87 @@ public void AddSessionObj(KServerSession serverSession, IpcService obj)
...
+                {
+                    Logger.Error?.Print(LogClass.Service, $"Failed to create thread on {Name}: {result}");
+                }
+            }
+
+            ServerLoop(0);
         }
 
-        private void ServerLoop()
+        // TODO: This would be better if we had a proper kernel lock primitive here
```
[the remark this answers is not in the document]
Comment (Thog) [reply Ryujinx.HLE/HOS/Services/ServerBase.cs:147]: That would lock up a HLE core

**Comment by Thog** [comment]
> I suppose
> 
> ```cs
>             // FIXME: We should support blocking operations.
>             // Right now they can't be supported because it would cause the
>             // service to lock up as we only have one thread processing requests.
>             flags |= EventFdFlags.NonBlocking;
> ```
> 
> can be removed now?

Quite true yeah, will remove that now 👍

File: Ryujinx.HLE/HOS/Services/ServerBase.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (gdkchan) [reply Ryujinx.HLE/HOS/Services/ServerBase.cs:147]: That seems odd, HLE threads are not assigned to cores. They are all free to run in parallel, although blocking syscalls still work with them.

[the remark this answers is not in the document]
Comment (marysaka) [reply Ryujinx.HLE/HOS/Services/ServerBase.cs]: I removed HLE threading for now

**Comment by marysaka** [comment]
> Fixes online play on Use Your Words (errors out trying to connect/hangs for minutes at a time on master 1.1.243) image

Probably not fixed anymore, will need retesting later (marking as outdated for now)

**Review by gdkchan — APPROVED** [review]
lgtm, shame that the fix with multiple threads had to be reverted.

**Pull request merged**